docs/reference/modules/node.asciidoc

```diff
@@ -41,6 +41,12 @@ A tribe node, configured via the `tribe.*` settings, is a special type  of
 coordinating only node that can connect to multiple clusters and perform
 search and other operations across all connected clusters.
 
+<<machine-learning-node,Machine learning node>>::
+A node that has `xpack.ml.enabled` and `node.ml` set to `true`, which is the
+default behavior when X-Pack is installed. If you want to use X-Pack machine
+learning features, there must be at least one machine learning node in your
+cluster.
+
```

Comment:
The coordinating node section below also needs to be updated to include ML nodes.